doc/source/basics.rst

```diff
@@ -71,8 +71,10 @@ the **array** property
    s.array
    s.index.array
 
-Depending on the data type (see :ref:`basics.dtypes`), :attr:`~Series.array`
-be either a NumPy array or an :ref:`ExtensionArray <extending.extension-type>`.
+:attr:`~Series.array` will always be an :class:`~pandas.api.extensions.ExtensionArray`.
+The exact details of what an ``ExtensionArray`` is and why pandas uses them is a bit
```

Comment:
Reading this, I am wondering if we should reconsider the name "ExtensionArray", if it is not only any more to "extend" pandas, but is actually also returned by all dtypes of pandas itself .. (but anyway, discussion for another issue :-))

Reply:
Pandas Array Interface and BasePandasArray so a lot better now :) But again, for a different discussion.